### DISCUSSION TITLE
RegEx usage

### DISCUSSION THREAD

**Discussion by Firecul** [ThioJoe/YT-Spammer-Purge#842] [Q&A]
Does anyone here use regex to scan for comments? 
I just tried it today and it's returning results for things it shouldn't as far as I know, basically all the comments. What am I missing?

I was using a video from a recent issue form `https://www.youtube.com/watch?v=-oVhr-Hd4MI` and trying to scan for any `.uno` websites. I have tried `["\w+\.uno"gi]` and `["[a-zA-Z]+\.uno"ig]` (with and without the outside `[]` to little success. It's returning results that don't have a match in addition to the ones that do
![image](https://user-images.githubusercontent.com/34950371/170897033-3a4a72f4-2caf-43b3-b86f-e6ccb4f2fd5a.png)

According to [Regex101.com](https://regex101.com/r/IZtqpU/1) it should work
![image](https://user-images.githubusercontent.com/34950371/170897359-131479d9-6e54-4d40-8d95-18a3c6121b69.png)

**Comment by Andre-4949** [comment]
![image](https://user-images.githubusercontent.com/83180381/171492204-b5ead631-0db6-4e2a-80c5-4672d7fba3c6.png)

If you want to *match* a given string with re.compile(...).match(...), the whole string has to match the regex - just modify it a bit and it should work. Just like here: 
`.*?(?P<link>\w+\.uno).*` or `.*?\w+\.uno.*`

⏫Please choose wether you need the link or not, i wasn't sure about it

I hope this helps in any kind, i didn't completly understand the question / problem but i thought i might want to help.

**Reply by Firecul** [reply]
I'll give it a try shortly and see if that works better

**Reply by Firecul** [reply]
Forgot all about this, unfortunately, your expression didn't return any results. At least it's more discriminating than mine and it does return the whole comment on regex101.

**Comment by ThioJoe** [comment]
Are you including those actual quotation marks in the expression? I don't think they should be there. Also the `ig` are flags that I think are separate. Hard to explain but basically the way python takes in the expression you don't put those flags. 

I believe you'd want to literally just put the expression as 
`\w+\.uno`

I tried it with brackets around it and it just flagged literally everything, so maybe that's a problem and I shouldn't have the example have brackets around it

**Reply by Firecul** [reply]
I have tried with and without quotation marks and with and without brackets, with no success. 
I was hoping someone would have an expression they know works so I could adapt it.
Even something as broad as `\S+\.uno` returns nothing.
![image](https://user-images.githubusercontent.com/34950371/172274800-9b81a8b5-ab35-41d4-b922-a960d6453939.png)
